src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+            "fuel_cost_per_mmbtu",
+        ]
+
+        def label_header_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label header rows.
+
+            This function labels rows it believes are possible headers based on whether
+            they contain information in certain key columns. Of those possible headers,
+            ones that contain a specific key word or phrase are dubbed headers. Leftover
+            possible header rows are evaluated as possible note rows in the
+            _label_notes_rows() function.
+            """
+            logger.info(f"{self.table_id.value}: Labeling header rows")
+
+            # Label possible header rows (based on the nan cols specified above)
+            df.loc[
+                df.filter(not_header_if_cols_na).isna().all(1), "possible_header"
+            ] = True
+
+            # Label good header rows (based on whether they contain key strings)
+            possible_header = df["possible_header"]
+            good_header = df["plant_name_ferc1"].str.contains("|".join(header_strings))
+            not_bad = ~df["plant_name_ferc1"].str.contains("|".join(nonheader_strings))
+
+            df.loc[possible_header & good_header & not_bad, "row_type"] = "header"
+            df.loc[df["plant_name_ferc1"].isin(exceptions), "row_type"] = "header"
+
+            return df
+
+        def label_total_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label total rows."""
+            logger.info(f"{self.table_id.value}: Labeling total rows")
+
+            df.loc[df["plant_name_ferc1"].str.contains("total"), "row_type"] = "total"
+
+            # Now deal with some outliers:
+
+            # There are a couple rows that contain the phrase "amounts are for" in the
+            # plant name that contain total pertaining to total records above. This
+            # section of code moves the total row values that are reported as notes to
+            # correct row above and nulls the values in the notes row.
+            num_cols = [
+                x
+                for x in df.select_dtypes(include=["float", "Int64"]).columns.tolist()
+                if x not in ["utility_id_ferc1", "report_year", "license_id_ferc1"]
+            ]
+            bad_row = df[
+                (df["plant_name_ferc1"].str.contains("amounts are for"))
+                & (df["capacity_mw"] > 0)
+            ]
+
+            df.loc[bad_row.index, num_cols] = df.loc[bad_row.index - 1][num_cols].values
+            df.loc[bad_row.index - 1, num_cols] = np.nan
+
+            return df
+
+        def label_notes_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Remove clumps of consecutive rows flagged as possible headers.
+
+            FERC has lots of note rows that are not headers but are also not useful for
+            analysis. This function looks for rows flagged as possible headers (based on
+            NA values) and checks to see if there are multiple in a row. A header row is
```

Comment:
hahah yes a valid point. There are 750 instances of note rows that are readily mappable to record rows via a footnote indicator like (a) or (1). These help fill in the `license_id_ferc1` field and also help retain other useful information reported alongside the plants. Granted, not wildly useful information, but information none-the-less. If we don't want to retain this information, we don't have to! This was just part of the work I did last year, so I figured I'd keep it in and then we could decide to nix it or not.

Reply:
I'm thinking that it might save a lot of code/time to see if we can do this footnote mapping on non-header possible_header_or_note rows. I.e., This would avoid all note row mapping and would just look for rows with the prescribed NA values that are not headers that might have a footnote indicator in them.

Reply:
I think this is out of scope for now